ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogCache.java

```diff
@@ -178,12 +198,21 @@ long getTotalFileSize() {
       return sizeInBytes;
     }
 
-    long getTotalCacheSize() {
-      return segments.stream().mapToLong(LogSegment::getTotalCacheSize).sum();
+    LogSegmentListCacheInfo getCacheInfo() {
+      long totalCacheSize = 0L;
+      long cachedCount = 0L;
+      for (LogSegment segment: segments) {
+        if (segment.hasCache()) {
+          cachedCount++;
+        }
+        // TODO: Should we only increment when hasCache = true?
```

Comment:
Although the result should the same, it seems better to increment when hasCache = true.

Reply:
Sure, updated according to the suggested patch diff.